provider/equinix/environ.go

```diff
@@ -177,11 +249,173 @@ func newConfig(cfg, old *config.Config) (*environConfig, error) {
+			"apt-get update",
+			"DEBIAN_FRONTEND=noninteractive apt-get --option=Dpkg::Options::=--force-confdef --option=Dpkg::Options::=--force-confold --option=Dpkg::Options::=--force-unsafe-io --assume-yes --quiet install dmidecode snapd lxd",
+		)
+	}
+
+	userdata, err := providerinit.ComposeUserData(args.InstanceConfig, cloudCfg, EquinixRenderer{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// Render the required tags for the instance.
+	var packetTags []string
+	for k, v := range args.InstanceConfig.Tags {
+		packetTags = append(packetTags, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	hostname, err := e.namespace.Hostname(args.InstanceConfig.MachineId)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	device := &packngo.DeviceCreateRequest{
+		Hostname:     hostname,
+		Facility:     []string{e.cloud.Region},
+		Plan:         spec.InstanceType.Name,
+		OS:           spec.Image.Id,
+		ProjectID:    e.cloud.Credential.Attributes()["project-id"],
+		BillingCycle: "hourly",
+		UserData:     string(userdata),
+		Tags:         packetTags,
+	}
+
+	subnetIDs, err := e.getSubnetsToZoneMap(ctx, args)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var requestedPublicAddr, requestedPrivateAddr bool
+	if len(subnetIDs) != 0 {
+		logger.Debugf("requesting a machine with address in subnet(s): %v", subnetIDs)
+		for _, subnetID := range subnetIDs {
+			net, _, err := e.equinixClient.ProjectIPs.Get(subnetID, &packngo.GetOptions{})
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+
+			requestedPublicAddr = requestedPublicAddr || net.Public
+			requestedPrivateAddr = requestedPrivateAddr || !net.Public
+
+			// Packet requires us to request at least a /31 for IPV4
+			// addresses and a /127 for IPV6 ones.
+			cidrSize := 31
+			if net.AddressFamily != 4 {
+				cidrSize = 127
+			}
+
+			ipBlock := packngo.IPAddressCreateRequest{
+				AddressFamily: net.AddressFamily,
+				Public:        net.Public,
+				CIDR:          cidrSize,
+				Reservations:  []string{net.ID},
+			}
+			device.IPAddresses = append(device.IPAddresses, ipBlock)
+		}
+	}
+
+	// In order to spin up a new device, we must specify at least one
+	// public and one private address.
+	if !requestedPrivateAddr {
+		// Allocate a private address from the default address pool.
+		device.IPAddresses = append(device.IPAddresses, packngo.IPAddressCreateRequest{
+			Public:        false,
+			AddressFamily: 4,
+			CIDR:          31,
+		})
+	}
+	if !requestedPublicAddr {
+		// Allocate a public address from the default address pool.
+		device.IPAddresses = append(device.IPAddresses, packngo.IPAddressCreateRequest{
+			Public:        true,
+			AddressFamily: 4,
+			CIDR:          31,
+		})
+	}
+
+	d, _, err := e.equinixClient.Devices.Create(device)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	d, err = waitDeviceActive(e.equinixClient, d.ID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	inst := &equinixDevice{e, d}
+	amd64 := arch.AMD64
```

Comment:
This assumption is wrong. You should probably populate the arch from the spec you looked up at the top of the method.